src/Utilities/BoostHelpers.hpp

```diff
@@ -77,6 +77,9 @@ void pup(er& p, boost::variant<Ts...>& var) noexcept {  // NOLINT
   p | send_index;
   (void)std::initializer_list<char>{
       ::BoostVariant_detail::pup_helper<Ts>(index, p, var, send_index)...};
+  if(var.which() != send_index) {
+    ERROR("boost variant did not transmit successfully.");
```

Comment:
- I suggest "boost variant" -> "boost::variant" for clarity
- Can you help me understand this check? When would this go wrong (as in, what does it mean for the variant not to transmit successfully)? Presumably this only ever triggers when unpacking? Perhaps putting this in an "if p.isUnpacking" and adding a comment might help.

Reply:
I think I'll actually just drop this check from the commit -- I added it because I thought this would fail in certain edge cases, but now I'm pretty sure as written this check always passes.